### PR TITLE
[WIP] autoquant prototype

### PR DESCRIPTION
**Pull request by jerryzh168**

### CODE REVIEW

File: tutorials/quantize_vit/run_vit_b_quant.py [hunk 1]
```diff
@@ -37,15 +40,27 @@
     unwrap_tensor_subclass(model)
 
 # temporary workaround to recover the perf with quantized model under torch.compile
-torch.backends.mha.set_fastpath_enabled(False)
+# torch.backends.mha.set_fastpath_enabled(False)
+
+# model = torch.compile(model, mode='max-autotune')
+model(*inputs)
+model.finalize_autoquant()
```
Comment (cpuhrsch) [inline tutorials/quantize_vit/run_vit_b_quant.py:47]: why is this needed?

Comment (cpuhrsch) [reply tutorials/quantize_vit/run_vit_b_quant.py:47]: Or put differently, another option here could be using a context.

**Comment by jerryzh168** [comment]
@cpuhrsch I'm planning to create another PR that is trying to subgraph extraction based on Vasiliy's code btw: https://github.com/pytorch-labs/ao_benchmarks/tree/main/experimental/float8_subgraph_analysis
